lib/experimental/class-gutenberg-rest-template-revision-count.php

```diff
@@ -72,4 +72,29 @@ protected function prepare_revision_links( $template ) {
 
 		return $links;
 	}
+
+	/**
+	 * Checks if a given request has access to read templates.
+	 *
+	 * @since 6.5
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has read access, WP_Error object otherwise.
+	 */
+	public function get_items_permissions_check( $request ) {
+		/*
+		 * Allow access to anyone who can edit posts.
+		 */
+		if ( ! current_user_can( 'edit_posts' ) ) {
+			return new WP_Error(
+				'rest_cannot_manage_templates',
+				__( 'Sorry, you are not allowed to access the templates on this site.' ),
+				array(
+					'status' => rest_authorization_required_code(),
+				)
+			);
+		}
+
+		return true;
```

Comment:
I'm aware this isn't the right spot for this. Though since the template controler was already extended I'm not sure where else to put it